src/python/pants/backend/jvm/tasks/jvm_compile/zinc/zinc_compile.py

```diff
@@ -455,6 +451,10 @@ def _compile_hermetic(self, jvm_options, ctx, classes_dir, jar_file,
         ),
       ])
 
+    # If analysis file exists, then incremental compile is enabled.
```

Comment:
I don't think this is true; we use consistent filenames for these, and we unconditionally set this field on `CompileContext`...

Maybe we can modify the integration test you're adding to then modify a file again and do a compile with `--no-incremental` and check that it compiled both files again?

Reply:
A bit tricky: https://github.com/pantsbuild/pants/blob/e3cf637cc72ba10f34bd6eb2d11ac6242e819b72/src/python/pants/backend/jvm/tasks/jvm_compile/jvm_compile.py#L950-L955 runs before here, so it will actively delete the analysis file if the compile is not going to be incremental.

That said, the comment is in the best place, it should be in docstring of 
```
  def _compute_scratch_inputs(self, classes_dir, relpath_to_analysis, jar_file):
    """
    Compute for the scratch inputs for ExecuteProcessRequest.

    :param classes_dir: relative path to classes dir from buildroot
    :param relpath_to_analysis: relative path to zinc analysis file from buildroot
    :param jar_file: relative path to z.jar from buildroot
    :return: digest of merged analysis file and loose class files.
    """
    if not os.path.exists(relpath_to_analysis):
      return EMPTY_DIRECTORY_DIGEST
   ...
```
where checks for its existence.

Reply:
Adjusted.